### PR TITLE
Fix TreeView content mode

### PR DESCRIPTION
**Pull request by kaiguo**
[Internal issue](https://microsoft.visualstudio.com/OS/_workitems/edit/19782289)

Currently TreeView checks the content of ItemsSource to determine whether it's in "content mode" or "node mode". If a user set the binding to ItemsSource without initializing the data list, TreeView would be running in "node mode" and causes the later ItemsSource updates to fail.

Instead of checking the content/size of ItemsSource, now we set TreeView to "content mode" as soon as we see an ItemsSource property change event.

Test run: https://microsoft.visualstudio.com/WinUI/_build/results?buildId=13262319&view=ms.vss-test-web.test-result-details

### CODE REVIEW

File: dev/TreeView/TestUI/TreeViewLateDataInitTestPage.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        private ObservableCollection<TreeViewItemSource> m_testTreeViewItemsSource;
+        public ObservableCollection<TreeViewItemSource> TestTreeViewItemsSource
+        {
+            get { return m_testTreeViewItemsSource; }
+            set
+            {
+                if (m_testTreeViewItemsSource != value)
+                {
+                    m_testTreeViewItemsSource = value;
+                    NotifyPropertyChanged("TestTreeViewItemsSource");
```
Comment (kmahone) [inline dev/TreeView/TestUI/TreeViewLateDataInitTestPage.xaml.cs]: You can use nameof here to make things a bit more strongly typed.

Comment (kaiguo) [reply dev/TreeView/TestUI/TreeViewLateDataInitTestPage.xaml.cs]: I got an error saying nameof operator is a C# 6 feature, we are currently on 5, are we ok upgrading to 6?

Comment (ranjeshj) [reply dev/TreeView/TestUI/TreeViewLateDataInitTestPage.xaml.cs]: @llongley hit some issues there. Would be nice to move to the latest version.

Comment (chrisglein) [reply dev/TreeView/TestUI/TreeViewLateDataInitTestPage.xaml.cs]: I hit this as well in late October. There wasn't a reason to not do it anymore assuming the OS build still functions (and it'll noisily fail if it isn't supported). Edit the LangVersion to give it a try if you like.

File: dev/TreeView/TreeView.cpp [hunk 2]
```diff
@@ -246,8 +246,12 @@ void TreeView::OnItemsRemoved(int index, int count)
 
 void TreeView::SyncRootNodesWithItemsSource()
 {
-    bool isContentMode = m_itemsDataSource != nullptr;
-    winrt::get_self<TreeViewNode>(m_rootNode.get())->IsContentMode(isContentMode);
+    winrt::get_self<TreeViewNode>(m_rootNode.get())->IsContentMode(true);
```
[the remark this answers is not in the document]
Comment (kaiguo) [reply dev/TreeView/TreeView.cpp]: The default value is false and we set it true once there is an ItemsSource change event. If a user works with TreeViewNode directly, the code won't go to this ItemsSource path and that value will be kept false.

File: dev/TreeView/TestUI/TreeViewLateDataInitTestPage.xaml.cs (hunk 1 above)
Comment (kaiguo) [reply dev/TreeView/TestUI/TreeViewLateDataInitTestPage.xaml.cs]: I updated to LangVersion to 7 and manually queued an os build https://microsoft.visualstudio.com/WinUI/WinUI%20Team/_build/results?buildId=13278936. Seems fine to me?

**Pull request merged**